src/System.Net.Mime/src/System/Net/Mime/MimeBasePart.cs

```diff
@@ -0,0 +1,289 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Specialized;
+using System.Text;
+using System.Net.Mail;
+
+namespace System.Net.Mime
+{
+    internal class MimeBasePart
+    {
+        internal const string DefaultCharSet = "utf-8";//"iso-8859-1";
+
+        protected ContentType _contentType;
+        protected ContentDisposition _contentDisposition;
+        private HeaderCollection _headers;
+
+        internal MimeBasePart() { }
+
+        internal static bool ShouldUseBase64Encoding(Encoding encoding) =>
+            encoding == Encoding.Unicode || encoding == Encoding.UTF8 || encoding == Encoding.UTF32 || encoding == Encoding.BigEndianUnicode;
```

Comment:
Is it possible that this could be called with some other instance of these encodings (e.g. `ShouldUseBase64Encoding(new UTF8Encoding())`), and if so, does the differing result matter?